packages/core/src/modal/modal.element.ts

```diff
@@ -46,7 +56,13 @@ import { styles } from './modal.element.css.js';
  * @cssprop --box-shadow
  * @cssprop --width
  */
-export class CdsModal extends CdsInternalOverlay {
+@animate({
+  hidden: {
+    true: reverseAnimation(AnimationModalEnterName),
+    false: AnimationModalEnterName,
+  },
+})
```

Comment:
Animations are not inherited because decorators are applied when a new class instance is created. This is probably the best argument for making the decorator a headless component.